cmd/datagen/datagen.go

```
@@ -0,0 +1,201 @@
+	} else if err := writeFeatures(*outPath, *fc); err != nil {
+		log.Fatal("error writing features: ", err)
+	} else if err := writeAttribution(*outPath, attributionFiles); err != nil {
+		log.Fatal("error writing attribution: ", err)
+	}
+}
+
+func readInputs(in []string, propsFileName string) (*geojson.FeatureCollection, error) {
+	var props *geojson.FeatureCollection
+	if propsFileName != "" {
+		md, err := readGeoJSON(propsFileName)
+		if err != nil {
+			return nil, fmt.Errorf("read props GeoJSON file: %w", err)
+		}
+		props = md
+	}
+
+	fc := &geojson.FeatureCollection{}
+	for _, f := range in {
+		s, err := readGeoJSON(f)
+		if err != nil {
+			return nil, fmt.Errorf("read input GeoJSON file: %w", err)
+		}
+		if props != nil {
+			if err := extendProps(s, props); err != nil {
+				return nil, fmt.Errorf("extend properties: %w", err)
+			}
+		}
+		fc.Features = append(fc.Features, s.Features...)
+	}
+
+	return fc, nil
+}
+
+func writeFeatures(outPath string, fc geojson.FeatureCollection) error {
+	f, err := os.Create(outPath)
+	if err != nil {
+		return fmt.Errorf("create output file: %w", err)
+	}
+	defer func() { _ = f.Close() }()
+
+	zw, err := zstd.NewWriter(f, zstd.WithEncoderLevel(zstd.SpeedBestCompression))
+	if err != nil {
+		return fmt.Errorf("create zstd writer: %w", err)
+	}
+	defer func() { _ = zw.Close() }()
+
+	dataset, err := rgeo.LoadGeoJSON(fc)
+	if err != nil {
+		return fmt.Errorf("load GeoJSON: %w", err)
+	}
+
+	if err := dataset.Encode(zw); err != nil {
+		return fmt.Errorf("encode dataset: %w", err)
+	}
+
+	// explicit flush so that zw.Close always succeeds
+	if err := zw.Flush(); err != nil {
+		return fmt.Errorf("flush: %w", err)
+	}
+
+	return nil
+}
+
+// readGeoJSON parses a GeoJSON file as geojson.FeatureCollection
+func readGeoJSON(path string) (*geojson.FeatureCollection, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("open file: %w", err)
+	}
+	defer func() {
+		_ = f.Close()
+	}()
```

lets condense that into a single line, like in every other `defer`'ed `Close()`

---

That would leave a warning in GoLand for an ignored error from Close()

---

I just meant a cosmetic change from the above 3 lines to the single line

`defer func() { _ = f.Close() }()`